### PR TITLE
[5.2] Added ability to remove a route from the RouteCollection.

### PR DESCRIPTION
**Pull request by garygreen**

### CODE REVIEW

File: src/Illuminate/Routing/RouteCollection.php [hunk 1]
```diff
@@ -99,6 +99,70 @@ protected function addLookups($route)
     }
 
     /**
+     * Remove the route from the collection.
+     *
+     * @param  Route  $route
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/RouteCollection.php]: cs

File: src/Illuminate/Routing/RouteCollection.php [hunk 2]
```diff
@@ -99,6 +99,70 @@ protected function addLookups($route)
...
+        $this->removeFromCollections($route);
+    }
+
+    /**
+     * Remove the route from any look-up tables if necessary.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @return void
+     */
+    protected function removeLookups($route)
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/RouteCollection.php]: why no typehint but there is in the other function you added?

File: src/Illuminate/Routing/RouteCollection.php [hunk 3]
```diff
@@ -99,6 +99,70 @@ protected function addLookups($route)
...
+        }
+    }
+
+    /**
+     * Remove a route from the name list.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @return void
+     */
+    protected function removeFromNameList($route)
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/RouteCollection.php]: same

**Comment by GrahamCampbell** [comment]
What use case do you actually have for this please?

**Comment by garygreen** [comment]
In case you want to remove a route dynamically after it's been registered e.g. a package registers a route you do not want.

**Comment by GrahamCampbell** [comment]
> In case you want to remove a route dynamically after it's been registered e.g. a package registers a route you do not want.

Massive :-1: from me I'm afraid. You should stop the package registering such routes.

File: src/Illuminate/Routing/RouteCollection.php (hunk 2 above)
Comment (garygreen) [inline src/Illuminate/Routing/RouteCollection.php]: The other lookup functions didnt type hint, just copied those but will change.

**Comment by garygreen** [comment]
Someone else on slack was asking about a remove function as well, maybe they have another use case.

**Comment by GrahamCampbell** [comment]
> Symphony router supports deleting a route as well.

Doesn't mean it's a good idea though.

> I agree it's best to avoid registering those routes

Indeed.

**Pull request closed without merging**